airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -291,9 +300,12 @@ private void allocateMemoryToState(final long sizeInBytes) {
           throw new RuntimeException(e);
         }
       }
+      LOGGER.debug(getMemoryUsageMessage());
     }
-    memoryUsed.addAndGet(sizeInBytes);
-    LOGGER.debug("State Manager memory usage: Allocated: {}, Used: {}, % Used {}",
+  }
+
+  public String getMemoryUsageMessage() {
+    return String.format("State Manager memory usage: Allocated: %s, Used: %s, percentage Used %f",
```

Comment:
I feel pretty strongly we should not be doing this - this is an expectation of what the memory should be but does not conform to what the JVM is doing with memory. In practice this means engineers debugging connections might see this statement and think its true when its an estimate. I would recommend at least one of the following changes:

- remove this method, and its call from `BufferManager` (at the very least, don't repeat the pattern) and remove lines 79-85 in `BufferManager`
- remove the calculated values (`memoryAllocated`, `memoryUsed`, maybe all the values from `GlobalMemoryManager` class and use the `Runtime.getRuntime().totalMemory()` and `Runtime.getRuntime().freeMemory()` instead

Reply:
I understand that it is confusing but the goal of this log is to give an estimation of what we are storing in the buffer and not the exact value of what the memory usage is. While debugging we used VisualVM in order to access to the total and free memory.
This estimation was wrong because of what this PR is aiming to fix.

Reply:
I am missing something - I think I don't understand why its useful to log an estimate? Especially when there are better resources available for memory usage? My concern is that this gives a false impression of reality. In a vacuum with no context of your project, I would (and did) assume this was accurately reporting memory usage. At the very least, this log line should be changed to be explicit that it does not reflect actual measurements. Something like `bytes expected to be written (not an accurate representation of memory usage)` thats maybe too verbose but I want to be clear what my objection is here.

Reply:
> I would (and did) assume this was accurately reporting memory usage.

We thought the same thing when we inherited this! Totally agree it's not ideal, however, after being in the code for a while, I think I understand the general context it was written in.

The goal, afaict, is to provide a bounded topic of queues (partitioned by stream), such that as we read messages from upstream we can appropriately backpressure before we run out of memory. The catch is we don't know the size of the messages we are enqueuing until runtime, so we can't use static array sizes or what have you. The GlobalMemoryManager and its queues are an attempt to provide this functionality, but they have some issues.
 
These are 
1) misleading naming
2) poor sizing of the available memory—it is way too large so it never backpressures (we probably need to size per dest)
3) the estimate size of the messages is off (we are trying to improve this)
4) general bugs

Our tack thus far has been to incrementally improve it, so the log messages have been useful in that regard.

Reply:
ok if the messages have some use, the minimum change here is to the text in the log lines. Making them explicitly clear what their purpose is so that there can be no confusion. To be blunt, in their current state these log lines are more harmful than good to someone not on this project.

Reply:
Updated

Reply:
I'd also argue we should rename the class / methods as well. Maybe in a follow-up PR to unblock @benmoriceau